File: vscode/samurai-agent/tests/agent/code_parser/fixtures/python/sample_enums.py
# Python Enum Test Fixture
# Tests enum element extraction

from enum import Enum, IntEnum, Flag, auto

class Status(Enum):
    ACTIVE = 'active'
    INACTIVE = 'inactive'
    PENDING = 'pending'

class HttpMethod(Enum):
    GET = 'GET'
    POST = 'POST'
    PUT = 'PUT'
    DELETE = 'DELETE'

class Priority(IntEnum):
    LOW = 1
    MEDIUM = 2
    HIGH = 3
    CRITICAL = 4

class Permission(Flag):
    READ = auto()
    WRITE = auto()
    EXECUTE = auto()
    ALL = READ | WRITE | EXECUTE

# Enum with methods
class Color(Enum):
    RED = 1
    GREEN = 2
    BLUE = 3

    def is_primary(self) -> bool:
        return self in (Color.RED, Color.GREEN, Color.BLUE)

File: vscode/samurai-agent/tests/agent/code_parser/fixtures/typescript/sample_enums.ts
// TypeScript Enum Test Fixture
// Tests enum element extraction

export enum Status {
    Active = 'active',
    Inactive = 'inactive',
    Pending = 'pending'
}

enum HttpMethod {
    GET,
    POST,
    PUT,
    DELETE
}

export enum Color {
    Red = '#FF0000',
    Green = '#00FF00',
    Blue = '#0000FF'
}

// Const enum (optimization)
const enum Direction {
    Up = 1,
    Down = 2,
    Left = 3,
    Right = 4
}

export { HttpMethod, Direction };

